lib/model/ble_device.dart

```diff
@@ -8,18 +8,37 @@ abstract class BleDevice {
   final String name;
   final DeviceIdentifier id;
   final BluetoothDevice bluetoothDevice;
+  DeviceCategory _category;
   BluetoothDeviceState bluetoothDeviceState = BluetoothDeviceState.disconnected;
 
-  bool get isConnected => bluetoothDeviceState == BluetoothDeviceState.connected;
+  bool get isConnected =>
+      bluetoothDeviceState == BluetoothDeviceState.connected;
 
-  BleDevice(this.name, this.id, this.bluetoothDevice, this.bluetoothDeviceState);
+  DeviceCategory get category => _category;
+
+  BleDevice(
+      this.name, this.id, this.bluetoothDevice, this.bluetoothDeviceState) {
+    _category = _nameToCategory(name);
+  }
 
   factory BleDevice.connected(String name, BluetoothDevice bluetoothDevice) {
     return ConnectedBleDevice(name, bluetoothDevice.id, bluetoothDevice);
   }
 
-  factory BleDevice.disconnected(String name, BluetoothDevice bluetoothDevice, FlutterBlue flutterBlue) {
-    return DisconnectedBleDevice(name, bluetoothDevice.id, bluetoothDevice, flutterBlue);
+  factory BleDevice.disconnected(
+      String name, BluetoothDevice bluetoothDevice, FlutterBlue flutterBlue) {
+    return DisconnectedBleDevice(
+        name, bluetoothDevice.id, bluetoothDevice, flutterBlue);
+  }
+
+  DeviceCategory _nameToCategory(String name) {
```

Comment:
IMHO it is responsibility of DeviceCategory to extract category base on a String

Reply:
But how? `DeviceCategory` is an `enum`, it cannot have methods. Do you suggest to transform it into a `class`?